flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -757,6 +775,64 @@ public void heartbeatFromResourceManager(final ResourceID resourceID) {
 		return CompletableFuture.completedFuture(executionGraph.getState());
 	}
 
+	//----------------------------------------------------------------------------------------------
+	// RestfulGateway RPC methods
+	//----------------------------------------------------------------------------------------------
+
+	@Override
+	public CompletableFuture<String> requestRestAddress(Time timeout) {
+		return restAddressFuture;
+	}
+
+	@Override
+	public CompletableFuture<AccessExecutionGraph> requestJob(JobID jobId, Time timeout) {
+		if (Objects.equals(jobGraph.getJobID(), jobId)) {
```

Comment:
When I see `Objects.equals`, I am assuming that it's possible that both arguments can be null. However, `jobGraph.getJobID()` is always non-null.

Reply:
True. Will refactor it.